### PR TITLE
trunner: phoenixd usb lookup

### PR DESCRIPTION
**Pull request by maska989**
## Description

trunner: check the solution using vid, pid instead of device name for plo

## Motivation and Context

JIRA: CI-304

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## How Has This Been Tested?

- [x] Already covered by automatic testing.
- [ ] New test added: (add PR link here).
- [ ] Tested by hand on: (list targets here).

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing linter checks and tests passed.
- [x] My changes generate no new compilation warnings for any of the targets.

## Special treatment

- [ ] This PR needs additional PRs to work (list the PRs, preferably in merge-order).
- [ ] I will merge this PR by myself when appropriate.

### CODE REVIEW

File: trunner/tools/phoenix.py [hunk 1]
```diff
@@ -22,6 +22,23 @@ def wait_for_dev(port, timeout=0):
             raise TimeoutError
 
 
+def wait_for_vid_pid(port_hint: str, timeout=0):
```
Comment (mateusz-bloch) [inline trunner/tools/phoenix.py:25]: I would renaming it to be more descriptive of its function.
Maybe something like: `find_device_by_vid_pid`

File: trunner/tools/phoenix.py [hunk 2]
```diff
@@ -22,6 +22,23 @@ def wait_for_dev(port, timeout=0):
...
+    asleep = 0
+    port = None
+
+    while port is None:
```
Comment (mateusz-bloch) [inline trunner/tools/phoenix.py:29]: I thinking if it's worth changing it to `while not port,` for better readability, but then the loop would check if it's not just None, but if it's not every state that python considers falsy

File: trunner/tools/phoenix.py [hunk 3]
```diff
@@ -22,6 +22,23 @@ def wait_for_dev(port, timeout=0):
...
+        time.sleep(0.01)
+        asleep += 0.01
+
+        for p in list_ports.grep(port_hint):
```
Comment (nalajcie) [inline trunner/tools/phoenix.py:33]: grep uses regexp as a param, either make a regexp or search for vid/pid by looping (they are properties of [ListPortInfo](https://pyserial.readthedocs.io/en/latest/tools.html#serial.tools.list_ports.ListPortInfo)).

Also - don't iterate over results (if there is more than one it should be considered an error?)

File: trunner/tools/phoenix.py [hunk 4]
```diff
@@ -22,6 +22,23 @@ def wait_for_dev(port, timeout=0):
...
+            port = p
+
+        if timeout and asleep >= timeout:
+            raise PhoenixdError(f"Couldn't find USB device with vid/pid: '{port_hint}'. Timeout.") from TimeoutError
```
Comment (nalajcie) [inline trunner/tools/phoenix.py:37]: this is a generic function (which might be reused not only in `Phoenixd` class so it should throw generic exception `TimeoutError`, not phoenixd-specific one

File: trunner/tools/phoenix.py (hunk 3 above)
Comment (maska989) [reply trunner/tools/phoenix.py]: I'm not 100% sure about the solution I used, so I'll wait for your feedback

File: trunner/tools/phoenix.py [hunk 5]
```diff
@@ -22,6 +22,30 @@ def wait_for_dev(port, timeout=0):
...
+def wait_for_vid_pid(port_hint: str, timeout=0):
+    """Searcher for connected usb device via vendor & product id"""
+
+    asleep = 0
+    port = None
+
+    while port is None:
+        time.sleep(0.01)
+        asleep += 0.01
+
```
Comment (nalajcie) [inline trunner/tools/phoenix.py:34]: i would go for specifying pid/vid as a number and than:

```python
found_ports = []

while not found_ports:
  time.sleep(0.01)
  asleep += 0.01

  found_ports = [ port for port in list_ports.comports() if port.pid == pid and port.vid == vid ]
  if len(found_ports) > 1:
     raise Exception(...)

  # timeout handling here

return found_ports[0].device
```
(the code was not tested, I'm just presenting a more robust approach)

File: trunner/tools/phoenix.py [hunk 6]
```diff
@@ -100,11 +123,7 @@ def _reader(self):
         self.proc.expect(pexpect.EOF, timeout=None)
 
     def _run(self):
-        try:
-            wait_for_dev(self.port, timeout=10)
-        except TimeoutError as exc:
-            raise PhoenixdError(f"couldn't find {self.port}") from exc
-
+        self.port = wait_for_vid_pid(0x16F9, 0x0003, timeout=10)
```
Comment (nalajcie) [inline trunner/tools/phoenix.py:126]: - probably you should catch exception(s) and translate them to `PhoenixdError` to remain compatible with the current API
- `wait_for_dev` is now dead code, should be removed

File: trunner/tools/phoenix.py [hunk 7]
```diff
@@ -22,6 +22,30 @@ def wait_for_dev(port, timeout=0):
             raise TimeoutError
 
 
+def wait_for_vid_pid(vid: int, pid: int, timeout=0):
+    """Searcher for connected usb device via vendor & product id"""
```
Comment (nalajcie) [inline trunner/tools/phoenix.py:26]:
```suggestion
    """wait for connected usb serial device with required vendor & product id"""
```

File: trunner/tools/phoenix.py [hunk 8]
```diff
@@ -101,9 +115,9 @@ def _reader(self):
 
     def _run(self):
         try:
-            wait_for_dev(self.port, timeout=10)
-        except TimeoutError as exc:
-            raise PhoenixdError(f"couldn't find {self.port}") from exc
+            self.port = wait_for_vid_pid(self.vid, self.pid, timeout=1)
```
Comment (nalajcie) [inline trunner/tools/phoenix.py:118]: You've changed timeout from 10s to 1s, is this intended? Maybe make it a separate commit so git log would clearly show that not only the lookup mechanism has changed but also the timeout itself.

Comment (maska989) [reply trunner/tools/phoenix.py]: Sorry, that was a mistake, I used this timeout to speed up debugging. We probably want to wait this 10 sec. because there is an option that the USB port will raise slower after some changes.

**Review by nalajcie — APPROVED** [review]
note for the future: dot (`.`) at the end of the commit message title is not needed

**Review by damianloew — APPROVED** [review]
